src/functions/shouldUpdateTask.ts

```diff
@@ -0,0 +1,20 @@
+import { ITask } from "store/reducers/toDosListReducer";
+
+export const shouldUpdateTask = (
+  arr: Record<string, ITask & string>[],
+  day: string,
+  task: string,
+) => {
+  const findDay = arr.find(el => el.id === day);
+  if (findDay) {
+    const findToDo = Object.entries(findDay);
+    const findToDoFilter = findToDo
+      .map(el => el[1])
+      .filter(el => el.task !== undefined);
+    const elToDo = findToDoFilter.find(el => el.task === task);
```

Comment:
Ты мапаешь el => el[1], это незачем делать. Можно использовать Object.values сразу

```suggestion
    const findToDo = Object.values(findDay).filter(element => Boolean(element.task)).find(element => element.task === task);
  
```